### PR TITLE
Save singleR reference label and ontology data frame to singleR results

### PR DESCRIPTION
**Pull request by allyhawkins**
In working on processing some projects through CellAssign, I was running into an error when adding the cell types to the SCE object. 
Screenshot 2023-10-30 at 11 48 39 AM

I went back and realized that we weren't properly storing the name of the reference label in the SingleR results file. Also, we didn't pass on the `cell_ontology_df` from the model to the results. We need that to map cell ontology IDs. 

One other thing to note is that I had to separate out modifying the `colData` and replacing the `colData`. Nextflow was complaining about the `colData` function.

### CODE REVIEW

File: bin/add_celltypes_to_sce.R [hunk 1]
```diff
@@ -95,10 +95,11 @@ if(!is.null(opt$singler_results)){
   }
 
   # add annotations to colData
-  colData(sce) <- colData(sce) |>
+  new_coldata <- colData(sce) |>
     as.data.frame() |>
     dplyr::left_join(annotations_df, by = c("barcodes")) |>
     DataFrame(row.names = colData(sce)$barcodes)
+  colData(sce) <- new_coldata
```
Comment (sjspielman) [inline bin/add_celltypes_to_sce.R:102]: i dont understand why nextflow would complain about this other than it likes to complain, but sure why not.

File: bin/classify_SingleR.R [hunk 2]
```diff
@@ -95,7 +95,10 @@ singler_results <- SingleR::classifySingleR(
 # add reference name to singler_results DataFrame metadata
 metadata(singler_results)$reference_name <- reference_name
 # add label name to metadata
-metadata(singler_results)$reference_label <- metadata(singler_model)$reference_label
+metadata(singler_results)$reference_label <- singler_model$reference_label
+# save cell ontology table to results
+# if this doesn't exist it will just be NULL
```
Comment (sjspielman) [inline bin/classify_SingleR.R:100]: we probably dont need this comment although it is true! i dont mind leaving it though if you really want it!

**Review by sjspielman — APPROVED** [review]
👍

File: bin/classify_SingleR.R (hunk 2 above)
Comment (allyhawkins) [reply bin/classify_SingleR.R:100]: I think this is more a reminder to myself if things break later and I'm trying to figure out why.

File: bin/add_celltypes_to_sce.R (hunk 1 above)
Comment (allyhawkins) [reply bin/add_celltypes_to_sce.R:102]: For context, this was the error I was seeing, but could not replicate it locally. 

Screenshot 2023-10-30 at 12 35 33 PM

**Pull request merged**